locale/core.yml

```diff
@@ -225,10 +225,14 @@ core:
       delete_forever_button: => core.ref.delete_forever
       log_in_to_reply_button: Log In to Reply
       rename_button: Rename
-      rename_text: "Enter a new title for this discussion:"
       reply_button: => core.ref.reply
       restore_button: => core.ref.restore
 
+      # These translations are used by the rename discussion modal.
+      rename_modal:
```

Comment:
It's a good idea to separate these out, but I'd suggest "rename_discussion:" as the key.

(Please compare the keys for other modal discussions.)